src/styles/colorData.ts

```diff
@@ -0,0 +1,103 @@
+import { brandAlt, neutral } from '@guardian/source-foundations';
+import type { Extras, ColorValueHex } from '../logic/types';
+
+const buttonStyles = {
+    textPrimary: neutral[7],
+    backgroundPrimary: brandAlt[400],
+    backgroundPrimaryHover: brandAlt[300],
+    textSecondary: neutral[7],
+    backgroundSecondary: neutral[93],
+    backgroundSecondaryHover: neutral[86],
+    borderSecondary: neutral[86],
+};
+
+export const contributionsTheme = {
+    button: buttonStyles,
+    link: buttonStyles,
+};
+
+export interface LoadingDotsColorStyles {
+    styleReminderAnimation: ColorValueHex;
+}
+
+export interface ReminderButtonColorStyles extends LoadingDotsColorStyles {
+    styleReminderButton: ColorValueHex;
+    styleReminderButtonBackground: ColorValueHex;
+    styleReminderButtonHover: ColorValueHex;
+}
+
+export interface BannerColorStyles {
+    styleBackground: ColorValueHex;
+    styleHeader: ColorValueHex;
+    styleBody: ColorValueHex;
+    styleHighlight: ColorValueHex;
+    styleHighlightBackground: ColorValueHex;
+    styleButton: ColorValueHex;
+    styleButtonBackground: ColorValueHex;
+    styleButtonHover: ColorValueHex;
+    styleClose: ColorValueHex;
+    styleCloseBackground: ColorValueHex;
+    styleCloseHover: ColorValueHex;
+}
+
+export interface StyleableBannerColorStyles extends ReminderButtonColorStyles, BannerColorStyles {}
+
+// This will become an interface once we build a more generic newsletter epic with limited styling around the newsletter 1-click signup button
+export type EpicColorStyles = ReminderButtonColorStyles;
+
+interface AllAvailableColorStyles
+    extends LoadingDotsColorStyles,
+        ReminderButtonColorStyles,
+        BannerColorStyles,
+        EpicColorStyles {}
+
+const colorStringStyles = [
+    'styleBackground',
+    'styleHeader',
+    'styleBody',
+    'styleHighlight',
+    'styleHighlightBackground',
+    'styleButton',
+    'styleButtonBackground',
+    'styleButtonHover',
+    'styleReminderButton',
+    'styleReminderButtonBackground',
+    'styleReminderButtonHover',
+    'styleReminderAnimation',
+    'styleClose',
+    'styleCloseBackground',
+    'styleCloseHover',
+];
+
+type ColorStylesType = keyof AllAvailableColorStyles;
+
+export function getColors(userVals: Extras, defaults: Partial<AllAvailableColorStyles>) {
```

Comment:
I wonder if using generics here would be a better way to describe the interface of `getColours`? I don't think `getColors` would ever be called with an object which had properties of both `BannerColorStyles` and `EpicColorStyles`, hence the need for `Partial`. If it was generic it called be called concretely with the specific relevant type acknowledging the fact that it would be called with a specific concrete type:

```
function getColors(userVals: Extras, defaults: T): T {
  // ...
}

// then called like:
getColors<BannerColorStyles>(userVals, defaultBannerColours)
getColors<EpicColorStyles>(userVals, defaultEpicColors)
/// etc
```

Reply:
We've agreed to look at this in a future PR